docs: list minimal permissions for Constellation setup (#1442)
edgelesssys/constellation: docs/docs/getting-started/install.md

By default, Constellation tries to register these automatically if they haven't been registered before.

To [automatically create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
* `Microsoft.Authorization/roleDefinitions/*`
* `Microsoft.Authorization/roleAssignments/*`

> * `*/register/action` [1]

Reviewer: ```suggestion
* `*/register/action` \[1]
```
nit: better escape the opening bracket

* `Microsoft.ManagedIdentity/userAssignedIdentities/*`

> * `Microsoft.Resources/subscriptions/resourcegroups/*`
> 
> To [create a Constellation cluster](../workflows/create.md#the-create-step), you need the following permissions:
> * `Microsoft.Insights/components/*`
> * `Microsoft.Network/publicIPAddresses/*`
> * `Microsoft.Network/virtualNetworks/*`
> * `Microsoft.Network/loadBalancers/*`
> * `Microsoft.Network/networkSecurityGroups/*`
> * `Microsoft.Network/loadBalancers/backendAddressPools/*`
> * `Microsoft.Network/virtualNetworks/subnets/*`
> * `Microsoft.Compute/virtualMachineScaleSets/*`
> * `Microsoft.ManagedIdentity/userAssignedIdentities/*`

Reviewer: ```suggestion
* `Microsoft.Resources/subscriptions/resourcegroups/*`

The built-in `Owner` role is a superset of these permissions.

To [create a Constellation cluster](../workflows/create.md#the-create-step), you need the following permissions:
* `Microsoft.Insights/components/*`
* `Microsoft.Network/publicIPAddresses/*`
* `Microsoft.Network/virtualNetworks/*`
* `Microsoft.Network/loadBalancers/*`
* `Microsoft.Network/networkSecurityGroups/*`
* `Microsoft.Network/loadBalancers/backendAddressPools/*`
* `Microsoft.Network/virtualNetworks/subnets/*`
* `Microsoft.Compute/virtualMachineScaleSets/*`
* `Microsoft.ManagedIdentity/userAssignedIdentities/*`

The built-in `Contributor` role is a superset of these permissions.
```
Should we give the user simpler alternatives? (Although Owner is not that cool of course.)
If so, do we have something similar on GCP?

Author: I think we can figure out the needed roles for GCP as well. But, as you already mentioned, they might be very extensive

Follow Microsoft's guide on [understanding](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-definitions) and [assigning roles](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments).

> 1: `*/register/Action` can be omitted if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration

Reviewer: ```suggestion
1: You can omit `*/register/Action` if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration.
```
nit: prefer active voice

By default, Constellation tries to register these automatically if they haven't been registered before.

> To [automatically create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:

Reviewer: ```suggestion
To [create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```
Nils agreed to revert this https://github.com/edgelesssys/constellation/pull/1442#discussion_r1147527831

* `Microsoft.Authorization/roleDefinitions/*`
* `Microsoft.Authorization/roleAssignments/*`
* `*/register/action` \[1]
* `Microsoft.ManagedIdentity/userAssignedIdentities/*`
* `Microsoft.Resources/subscriptions/resourcegroups/*`

> The built-in `Owner` role is a superset of these permissions.

Reviewer: > Any way to silence the review-🐶 here?

add it to docs/styles/Vocab/constellation/accept.txt

By default, Constellation tries to register these automatically if they haven't been registered before.

> To [create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:

Reviewer 2: ```suggestion
To [automatically create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```
To highlight that this refers to our feature so we don't confuse people who want to set these up manually.

Reviewer: But doesn't manually creating it require the same permissions?

Reviewer 2: True. Let’s just keep it as it is then 😄

* `Microsoft.Authorization/roleDefinitions/*`
* `Microsoft.Authorization/roleAssignments/*`

> * `*/register/action` (can be ommitted if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration)

Reviewer 2: I would make this a footnote maybe after and explain this after the list of all permissions. I think that may look cleaner in the docs eventually.

* `Microsoft.Network/virtualNetworks/subnets/*`
* `Microsoft.Compute/virtualMachineScaleSets/*`
* `Microsoft.ManagedIdentity/userAssignedIdentities/*`
* `Microsoft.Attestation/attestationProviders/*` \[2]

The built-in `Contributor` role is a superset of these permissions.

Follow Microsoft's guide on [understanding](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-definitions) and [assigning roles](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments).

1: You can omit `*/register/Action` if the resource providers mentioned above are already registered and the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable is set to `true` when creating the IAM configuration.

> 2: You can omit `Microsoft.Attestation/attestationProviders/*` if `EnforceIDKeyDigest` is set to `MAAFallback` in the [config file](../workflows/config.md#configure-your-cluster).

Reviewer 2: ```suggestion
2: You can omit `Microsoft.Attestation/attestationProviders/*` if `EnforceIDKeyDigest` is not set to `MAAFallback` in the [config file](../workflows/config.md#configure-your-cluster).
```

To set up a Constellation cluster, you need to perform two tasks that require permissions: create the infrastructure and create roles for cluster nodes. Both of these actions can be performed by different users, e.g., an administrator to create roles and a DevOps engineer to create the infrastructure.

> To create the IAM configuration for Constellation, you need the following permissions:

Reviewer: ```suggestion
To [create the IAM configuration](../workflows/config.md#creating-an-iam-configuration) for Constellation, you need the following permissions:
```
